pkg/resources/reconciler.go

```diff
@@ -179,15 +191,23 @@ func (r *Reconciler) ReconcileFinalizer(ctx context.Context, client k8sclient.Cl
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
-func (r *Reconciler) ReconcilePullSecret(ctx context.Context, namespace, secretName string, inst *integreatlyv1alpha1.RHMI, client k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
-	pullSecretName := DefaultOriginPullSecretName
-	if secretName != "" {
-		pullSecretName = secretName
+func (r *Reconciler) ReconcilePullSecret(ctx context.Context, destSecretNamespace, destSecretName string, inst *integreatlyv1alpha1.RHMI, client k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+	srcSecretName := ""
+	srcSecretNamespace := ""
+	if inst.Spec.PullSecret.Name == "" {
+		srcSecretName = DefaultOriginPullSecretName
+	} else {
+		srcSecretName = inst.Spec.PullSecret.Name
+	}
+	if inst.Spec.PullSecret.Namespace == "" {
+		srcSecretNamespace = DefaultOriginPullSecretNamespace
+	} else {
+		srcSecretNamespace = inst.Spec.PullSecret.Namespace
 	}
```

Comment:
Is this logic correct? If we are explicitly copying the default pull secret in the reconcilers that require it, do we still need to do this here?

Reply:
So it shouldn't get to there because we check if the name is empty but I didn't think it hurts to leave in a default. 

I suppose it will error out if it doesn't find the secret it expects which is a better way to go. I'll remove it